client/browser-app/test/test.ts

```diff
@@ -0,0 +1,571 @@
+/*******************************************************************************
+ * Copyright (c) 2019 EclipseSource and others.
+ *
+ *   This program and the accompanying materials are made available under the
+ *   terms of the Eclipse Public License v. 2.0 which is available at
+ *   http://www.eclipse.org/legal/epl-2.0.
+ *
+ *   This Source Code may also be made available under the following Secondary
+ *   Licenses when the conditions for such availability set forth in the Eclipse
+ *   Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ *   with the GNU Classpath Exception which is available at
+ *   https://www.gnu.org/software/classpath/license.html.
+ *
+ *   SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ******************************************************************************/
+import { Selector } from "testcafe";
+
+// Insert the Path to the default Workbench here
+const absPathToWorkspace = "D:/Users/Uni/Dokumente/workspaces/EclipseSource/ecore-glsp/client/workspace";
```

Comment:
> To run the tests the path to the default Workspace needs to be set at the top of the file.

Absolute paths must absolutely be avoided, as it is not possible for each developer to adjust the path to his own local setup. We should maybe consider passing the workspace path programmatically, as an argument, when running the test (Something similar to the start:debug script, with `--root=../workspace` 

Even `localhost:3000` is probably an unreliable value, as it's just the default port on which the server is started, when nothing else is specified

Reply:
I understand the problem. But i am not quite sure how to handle it. Right now there seems to be no way to create custom parameters for testcafe tests. Do you have an idea on how we could work around that?

Reply:
Maybe we could simply resolve the relative path "../../workspace" with the help of the `path` library? Similar to how the jarpath is resolved in the coffee editor:

https://github.com/eclipsesource/coffee-editor/blob/73db58f116cd75967a663d0a758545ef484031f9/web/coffee-server/src/node/backend-module.ts#L39

Reply:
I made the Path relative. However i am not quite sure how and if we should handle the problem with the port, that @CamilleLetavernier mentioned.

Reply:
how would one start the tests? must this be added to the package.json ? can we pass the port there?

Reply:
To run the tests you run `testcafe chrome browser-app/test/test.ts´ in the client folder. 

As far as i understand it is currently not possible to pass custom parameters to testcafe.

Reply:
Hmm, I found this:
https://devexpress.github.io/testcafe/faq/#how-do-i-work-with-configuration-files-and-environment-variables
and 
https://stackoverflow.com/questions/57653002/how-to-pass-a-parameter-to-world-parameters-or-npm-run-command-to-be-consumed

So we could use either environment variables or the config. I like the solution combining both this way we could use an environment variable on the build server to set the port for theia and testcafe.
@CamilleLetavernier  what do you think?

Reply:
Having "universal default" values (e.g. in config.json) that we can optionally override with Environment Variables (On CI) would be ideal. I would avoid forcing developers to set custom environment variables if possible (But that's perfectly fine on CI)

If having universal default values (that work for everyone without any extra custom configuration) is not possible, then Environment Variables are OK.

So, for the path, relative path is fine; for the port, I would use a default of :3000 that we can override with an optional Environment Variable.

Reply:
Should this be implemented with this PR or should this be part of the integration with CI? Since we want to avoid that developers have to set a custom enviroment variable?

Reply:
I suggest to add this now, the integration with CI is not necessary for the fix as far as I see it.

Reply:
Ok. So I adjust the package.json that it uses lets say: `process.env.PORT` and if that is not set, it uses :3000. And for testcafe i create a config that contains the URL which then again uses the specified Port or the default value?

Reply:
no don't touch the package.json, just adapt the code to check for the variable and otherwise read the value from the config file as shown in the SO link. In the config you set the port to 3000 .

Reply:
Ok i did that. I did not use the config file i just created a const in the test, that takes the enviroment variable and if that is not available just calls 3000.

Reply:
Why not use the config file?

Reply:
The config file only works, if you call the run command in the directory the config is in, so in our case `browser-app/test`. This would prevent the developer from calling the tests in the client directory. If we are fine with that I can add it to the config file.

Reply:
I don't see this in the documentation. Aren't you simply importing the config file via a part in your test?